### DISCUSSION TITLE
What happens if I cannot remember my 4 digit PIN ?

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#1469] [Ideas]
We may need to discuss this workflow ahead of time 
CC @HenryRae @rowo @dubdabasoduba @AnnieMungai @pld

**Comment by AnnieMungai** [comment]
@f-odhiambo at the moment in the Liberia implementation the CHA needs to contact their supervisor who has administrative rights to change the pin.

**Reply by f-odhiambo** [reply]
Where is this reset initiated? On the supervisor app or on the web ?

**Reply by rowo** [reply]
I don't think we promised this being fixed by the CHSS (the direct supervisor) in the supervisor app or the web. Resetting on a different device wouldn't do anything because the CHA device is never connected to get that update.

I thought the idea was either:
1. The supervisor would be able to log in to the CHA app using an admin login process (accessed via the login page top right drop down and using an admin password) and set a new pin (I assume the prompt to set a pin is asked everytime an admin password is used).
2. The CHSS would actually know the CHA pins since they're set by LMH and would be able to provide the pin to the CHA.

**Reply by AnnieMungai** [reply]
Option 2 is correct @rowo for offline users.

The admin knows all the CHA pins because they set them. They can thus help the CHA remember. 

@f-odhiambo is the app currently able to allow an admin to change the pin while in an online mode?

**Reply by f-odhiambo** [reply]
No this is not yet implemented thus added this discussion to understand the workflow and then move into implementation

**Reply by dubdabasoduba** [reply]
@f-odhiambo The PIN implementation was not going to have the web do any PIN management. The PIN is currently saved in the device (on shared preferences) and not synced to the server or keycloak. 

There are two ways to reset the PIN
1. The Supervisor logs in using the CHWs username & password this allows them to reset the PIN for the CHW. -- Preferred way, this is the option communicated to implementors
2. Clearing of the app data clears the PIN  -- Not the prefered way. Should only be used as a backup

**Reply by dubdabasoduba** [reply]
https://docs.google.com/document/d/1Vc3i_EsN9-P8hk8Spyhx1GD9lodlQIj1IcspWLo7DWI/edit?usp=sharing This is the scoping documentation used for the feature. 

We can obviously iterate and improve the management pieces later on in the project
